library/ssl_tls13_generic.c

```diff
@@ -342,6 +342,19 @@ static int ssl_write_change_cipher_spec_coordinate( mbedtls_ssl_context* ssl )
         }
     }
 #endif /* MBEDTLS_SSL_SRV_C */
+
+#if defined(MBEDTLS_SSL_CLI_C) && defined(MBEDTLS_ZERO_RTT)
+    if( ssl->conf->endpoint == MBEDTLS_SSL_IS_CLIENT )
+    {
```

Comment:
Agreed! I missed covering the HRR case. As a side benefit, given the CCS coordinate function actually handles 3 distinct handshake states I think the switch statement makes the CCS semantics easier to follow.

Reply:
1. I opted to invert the conditional logic so defaulting `ret = SSL_WRITE_CCS_SKIP` isn't needed, but lmk if that was intentional in your example.
2. Even if allowed, is there any reason not to check `ssl->handshake->hello_retry_requests_received` so we only send one CCS in the HRR case?

Reply:
> 1. I opted to invert the conditional logic so defaulting `ret = SSL_WRITE_CCS_SKIP` isn't needed, but lmk if that was intentional in your example.

That's ok.

>     2. Even if allowed, is there any reason not to check `ssl->handshake->hello_retry_requests_received` so we only send one CCS in the HRR case?

Adding this would complicate the logic and add more code for a quite small gain. That's why I opted not to add it. But I do not have a strong opinion about it thus if you think it would be better with it, fine by me.

Reply:
Let's leave as-is.